kafka-rest/src/main/java/io/confluent/kafkarest/GroupMetadataObserver.java

```diff
@@ -0,0 +1,275 @@
+    properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, deserializer);
+    properties.putAll(appConfig.getConsumerProperties());
+
+    return new KafkaConsumer<>(properties);
+  }
+
+  private final Logger log = LoggerFactory.getLogger(GroupMetadataObserver.class);
+
+  private final KafkaRestConfig config;
+  private final AdminClientWrapper adminClientWrapper;
+
+  public GroupMetadataObserver(KafkaRestConfig config, AdminClientWrapper adminClientWrapper) {
+    this.config = config;
+    this.adminClientWrapper = adminClientWrapper;
+  }
+
+  /**
+   * <p>Get consumer group list</p>
+   *
+   * @return list of consumer groups
+   */
+  public List<ConsumerGroup> getConsumerGroupList(Option<Integer> offsetOpt,
+                                                  Option<Integer> countOpt) throws Exception {
+    if (Option.apply(adminClientWrapper).nonEmpty()) {
+      final boolean needPartOfData = offsetOpt.nonEmpty()
+              && countOpt.nonEmpty()
+              && countOpt.get() > 0;
+      Collection<ConsumerGroupListing> groupsOverview =
+              adminClientWrapper.listConsumerGroups();
+      if (needPartOfData) {
+        final Comparator<ConsumerGroupListing> consumerGroupListingComparator =
+                Comparator.comparing(ConsumerGroupListing::groupId);
+        final List<ConsumerGroupListing> consumerGroupListings = new ArrayList<>(groupsOverview);
+        consumerGroupListings.sort(consumerGroupListingComparator);
+        groupsOverview = JavaConverters.asJavaCollection(
+                JavaConverters.collectionAsScalaIterable(consumerGroupListings)
+                .slice(offsetOpt.get(), offsetOpt.get() + countOpt.get()));
+      }
+      final List<ConsumerGroup> result = new ArrayList<>();
+      for (ConsumerGroupListing eachGroupInfo : groupsOverview) {
+        final Node node = adminClientWrapper.describeConsumerGroups(
+                Collections.singleton(eachGroupInfo.groupId()))
+                .get(eachGroupInfo.groupId()).coordinator();
+        result.add(new ConsumerGroup(eachGroupInfo.groupId(),
+                new ConsumerGroupCoordinator(node.host(), node.port())));
+      }
+      return result;
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  /**
+   * <p>Get consumer group description</p>
+   *
+   * @param groupId - group name
+   * @return description of consumer group
+   */
+  public Set<Topic> getConsumerGroupTopicInformation(String groupId,
+                                                         Option<Integer> offsetOpt,
+                                                         Option<Integer> countOpt)
+          throws Exception {
+    if (Option.apply(adminClientWrapper).nonEmpty()) {
+      final Collection<MemberDescription> summariesOpt =
+              adminClientWrapper.describeConsumerGroups(Collections.singleton(groupId))
+                      .get(groupId).members();
+      if (summariesOpt.size() > 0) {
```

Comment:
Is this conditional needed? Won't it behave the same (i.e. return empty collection) if you just let it proceed here with empty summariesOpt? Each extra conditional added means additional cyclomatic complexity, making the code harder to understand.